lib/AppInfo/Application.php

```diff
@@ -20,12 +22,19 @@ public function __construct() {
 	}
 
 	public function register(IRegistrationContext $context): void {
+		// "This event is triggered when the files app is rendered. It can be used to add additional scripts to the files app."
+		// See: https://docs.nextcloud.com/server/latest/developer_manual/basics/events.html#oca-files-event-loadadditionalscriptsevent
+		$context->registerEventListener(\OCA\Files\Event\LoadAdditionalScriptsEvent::class, FilesLoadAdditionalScriptsListener::class);
+
+		// "Emitted before the rendering step of the public share page happens. The event holds a flag that specifies if it is the authentication page of a public share."
+		// See: https://docs.nextcloud.com/server/latest/developer_manual/basics/events.html#oca-files-sharing-event-beforetemplaterenderedevent
+		$context->registerEventListener(\OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent::class, PublicShareBeforeTemplateRenderedListener::class);
 	}
 
 	public function boot(IBootContext $context): void {
 		/**
 		 * Always add main script
 		 */
-		Util::addInitScript(self::APP_ID, 'openincryptpad-main', 'files');
+		// Util::addInitScript(self::APP_ID, 'openincryptpad-main', 'files');
```

Comment:
The `LoadAdditionalScriptsEvent` does work in older NC versions (we support) right? Than we can remove `boot` completely.